apps/omg_eth/test/omg_eth/ethereum_height_monitor_test.exs

```diff
@@ -0,0 +1,204 @@
+    end)
+  end
+
+  #
+  # Connection error
+  #
+
+  test "that the connection alarm gets raised and with EthereumConnectionError event when connection becomes unhealthy" do
+    # Initialize as healthy and alarm not present
+    _ = EthereumClientMock.set_faulty_response(false)
+    :ok = pull_client_alarm(100, [])
+    assert EthereumHeightMonitor.get_events() == {:ok, []}
+
+    # Toggle faulty response
+    _ = EthereumClientMock.set_faulty_response(true)
+
+    # Assert the alarm and event are present
+    assert pull_client_alarm(100,
+             ethereum_connection_error: %{node: :nonode@nohost, reporter: OMG.Eth.EthereumHeightMonitor}
+           ) == :ok
+
+    assert {:ok, [%Event.EthereumConnectionError{}]} = EthereumHeightMonitor.get_events()
+  end
+
+  test "that the connection alarm gets cleared and without EthereumConnectionError event when connection becomes healthy" do
+    # Initialize as unhealthy
+    _ = EthereumClientMock.set_faulty_response(true)
+
+    :ok =
+      pull_client_alarm(100,
+        ethereum_connection_error: %{node: :nonode@nohost, reporter: OMG.Eth.EthereumHeightMonitor}
+      )
+
+    assert {:ok, [%Event.EthereumConnectionError{}]} = EthereumHeightMonitor.get_events()
+
+    # Toggle healthy response
+    _ = EthereumClientMock.set_faulty_response(false)
+
+    # Assert the alarm and event are no longer present
+    assert pull_client_alarm(100, []) == :ok
+    assert EthereumHeightMonitor.get_events() == {:ok, []}
+  end
+
+  #
+  # Stalling sync
+  #
+
+  test "that the stall alarm gets raised and with EthereumStalledSync event when block height stalls" do
+    # Initialize as healthy and alarm not present
+    _ = EthereumClientMock.set_stalled(false)
+    :ok = pull_client_alarm(200, [])
+    assert EthereumHeightMonitor.get_events() == {:ok, []}
+
+    # Toggle stalled height
+    _ = EthereumClientMock.set_stalled(true)
+
+    # Assert alarm now present
+    assert pull_client_alarm(200,
+             ethereum_stalled_sync: %{node: :nonode@nohost, reporter: OMG.Eth.EthereumHeightMonitor}
+           ) == :ok
+
+    assert {:ok, [%Event.EthereumStalledSync{}]} = EthereumHeightMonitor.get_events()
+  end
+
+  test "that the stall alarm gets cleared and without EthereumStalledSync event when block height unstalls" do
+    # Initialize as unhealthy
+    _ = EthereumClientMock.set_stalled(true)
+
+    :ok =
+      pull_client_alarm(300, ethereum_stalled_sync: %{node: :nonode@nohost, reporter: OMG.Eth.EthereumHeightMonitor})
+
+    assert {:ok, [%Event.EthereumStalledSync{}]} = EthereumHeightMonitor.get_events()
+
+    # Toggle unstalled height
+    _ = EthereumClientMock.set_stalled(false)
+
+    # Assert alarm no longer present
+    assert pull_client_alarm(300, []) == :ok
+    assert EthereumHeightMonitor.get_events() == {:ok, []}
+  end
+
+  defp pull_client_alarm(0, _), do: {:cant_match, Alarm.all()}
+
+  defp pull_client_alarm(n, match) do
```

Comment:
not a big deal, but I think in other similar "countdown" cases, `n` went at the last position. Can't remember, where else this pattern was used